recipe/thirdparty/tpmodels/provider.go

```diff
@@ -0,0 +1,63 @@
+package tpmodels
+
+import "github.com/supertokens/supertokens-golang/supertokens"
+
+type ProviderInput struct {
+	ThirdPartyID string
+	Config       ProviderConfig
+	Override     func(provider *TypeProvider) *TypeProvider
+}
+
+type ProviderConfig struct {
+	Clients []ProviderClientConfig
+
+	AuthorizationEndpoint            string
+	AuthorizationEndpointQueryParams map[string]interface{}
+	TokenEndpoint                    string
+	TokenParams                      map[string]interface{}
+	ForcePKCE                        bool // Providers like twitter expects PKCE to be used along with secret
+	UserInfoEndpoint                 string
+	JwksURI                          string
+	OIDCDiscoveryEndpoint            string
+	UserInfoMap                      TypeUserInfoMap
+	ValidateIdTokenPayload           func(idTokenPayload map[string]interface{}, clientConfig ProviderConfigForClient) (bool, error)
```

Comment:
make this just return an error and nothing else. If the check fails, the user should just return an error with a message indicating why the check failed.

Reply:
Done